tests/lib/rules/no-class-fields.js

```diff
@@ -221,3 +223,36 @@ new RuleTester().run("no-class-fields", rule, {
         },
     ],
 })
+
+// -----------------------------------------------------------------------------
+// TypeScript
+// -----------------------------------------------------------------------------
+const parser = require.resolve("@typescript-eslint/parser")
+const tsconfigRootDir = path.resolve(__dirname, "../../fixtures")
+const project = "tsconfig.json"
+const filename = path.join(tsconfigRootDir, "test.ts")
+
+new RuleTester({ parser, parserOptions: { tsconfigRootDir, project } }).run(
+    `${ruleId} TS Full Types`,
+    rule,
+    {
+        valid: [{ filename, code: "class A { declare foo: string }" }],
```

Comment:
Hmm. Do you think we should also ignore the class in the following code?

```ts
declare class A { foo: string }
```

Reply:
Good idea!
I am not so familiar with eslint parser, but I guess `node.parent.parent` for PropertyDefinition should always be ClassDeclaration.